app/src/main/java/com/example/plantsapp/adapter/PlantsAdapter.kt

```diff
@@ -0,0 +1,35 @@
+package com.example.plantsapp.adapter
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import com.example.plantsapp.databinding.ItemPlantBinding
+import com.example.plantsapp.model.Plant
+
+class PlantsAdapter : ListAdapter<Plant, PlantViewHolder>(itemComparator) {
+
+    companion object {
+        val itemComparator = object : DiffUtil.ItemCallback<Plant>() {
+
+            override fun areItemsTheSame(oldItem: Plant, newItem: Plant): Boolean {
+                return oldItem.name == newItem.name
+            }
+
+            override fun areContentsTheSame(oldItem: Plant, newItem: Plant): Boolean {
+                return oldItem == newItem
+            }
+        }
+    }
+
```

Comment:
1. Instead of creating anonymous object - crate private class 
2. Consider moving this to other file instead of creating it inside of adapter